snapcraft/internal/project_loader/_config.py

```diff
@@ -86,12 +89,12 @@ def _remote_parts(self):
             self._remote_parts_attr = remote_parts.get_remote_parts()
         return self._remote_parts_attr
 
-    def __init__(self, project_options=None):
+    def __init__(self, project_options: project.Project=None) -> None:
         if project_options is None:
-            project_options = snapcraft.ProjectOptions()
+            project_options = project.Project()
 
-        self.build_snaps = set()
-        self.build_tools = []
+        self.build_snaps = set()  # type: Set[str]
+        self.build_tools = set()  # type: Set[str]
```

Comment:
Hold up, this is a larger change-- are you sure you want to change build_tools from a list to a set?

Reply:
As clarified offline, it already needs to be a set, ie. we use `|= set()` further down in the same constructor. But we can do even better, by not initializing it here since we're overwriting it anyway.